winml/test/adapter/AdapterSessionTest.cpp

```diff
@@ -0,0 +1,311 @@
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionLoadAndPurloinModel(session.get(), ort_model), ort_api);
+}
+
+void LoadAndPurloinModel() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+  LoadAndPurloinModel(session, "fns-candy.onnx");
+}
+
+void Initialize() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+
+  com_ptr<WinML::IModel> model;
+  const auto model_path = "fns-candy.onnx";
+  WINML_THROW_IF_FAILED(engine_factory->CreateModel(model_path, sizeof(model_path), model.put()));
+
+  Microsoft::WRL::ComPtr<WinML::IOnnxruntimeModel> onnxruntime_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(model->QueryInterface(IID_PPV_ARGS(&onnxruntime_model)));
+  OrtModel* ort_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(onnxruntime_model->DetachOrtModel(&ort_model));
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionLoadAndPurloinModel(session.get(), ort_model), ort_api);
+
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionInitialize(session.get()), ort_api);
+}
+
+class MockSink : public onnxruntime::logging::ISink {
+public:
+  static bool called;
+
+private:
+  void SendImpl(const onnxruntime::logging::Timestamp& timestamp, const std::string& logger_id, const onnxruntime::logging::Capture& message) override {
+    called = true;
+  }
+
+  void SendProfileEvent(onnxruntime::profiling::EventRecord&) const override { };
+};
+bool MockSink::called = false;
+
+void Profiling() {
+  ort_env->SetLoggingManager(nullptr);
+  const std::string logger_name("Default");
+  auto logging_manager = std::make_unique<onnxruntime::logging::LoggingManager>(
+    std::make_unique<MockSink>(),
+    onnxruntime::logging::Severity::kINFO,
+    false,
+    onnxruntime::logging::LoggingManager::InstanceType::Default,
+    &logger_name);
+  ort_env->SetLoggingManager(std::move(logging_manager));
+
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+
+  winml_adapter_api->SessionStartProfiling(ort_env, session.get());
+  LoadAndPurloinModel(session, "fns-candy.onnx");
+  winml_adapter_api->SessionEndProfiling(session.get());
+  WINML_EXPECT_TRUE(MockSink::called);
+}
+
+void CopyInputAcrossDevices() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(L"fns-candy.onnx", session_options);
+
+  constexpr std::array<int64_t, 4> dimensions{1, 3, 720, 720};
+  constexpr size_t input_tensor_size = [&dimensions]() {
+    size_t size = 1;
+    for (auto dim : dimensions)
+      size *= dim;
+    return size;
+  } ();
+
+  OrtMemoryInfo* memory_info;
+  THROW_IF_NOT_OK_MSG(ort_api->CreateCpuMemoryInfo(OrtArenaAllocator, OrtMemTypeDefault, &memory_info), ort_api);
+  std::vector<float> input_tensor_values(input_tensor_size);
+  OrtValue* input_tensor;
```

Comment:
test leaks ortvalue?

Reply:
I was going to run with leak checking after I got the crash figured out